examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,433 @@
+# We can now visualize the performance of the model with regards
+# to the 5th percentile, median and the 95th percentile:
+all_splits = list(ts_cv.split(X, y))
+train_idx, test_idx = all_splits[0]
+
+X_train, X_test = X.iloc[train_idx], X.iloc[test_idx]
+y_train, y_test = y.iloc[train_idx], y.iloc[test_idx]
+
+max_iter = 50
+gbrt_mean_poisson = HistGradientBoostingRegressor(loss="poisson", max_iter=max_iter)
+gbrt_mean_poisson.fit(X_train, y_train)
+mean_predictions = gbrt_mean_poisson.predict(X_test)
+
+gbrt_median = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.5, max_iter=max_iter
+)
+gbrt_median.fit(X_train, y_train)
+median_predictions = gbrt_median.predict(X_test)
+
+gbrt_percentile_5 = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.05, max_iter=max_iter
+)
+gbrt_percentile_5.fit(X_train, y_train)
+percentile_5_predictions = gbrt_percentile_5.predict(X_test)
+
+gbrt_percentile_95 = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.95, max_iter=max_iter
+)
+gbrt_percentile_95.fit(X_train, y_train)
+percentile_95_predictions = gbrt_percentile_95.predict(X_test)
+
+last_hours = slice(-96, None)
+fig, ax = plt.subplots(figsize=(15, 7))
+plt.title("Predictions by regression models")
+ax.plot(
+    y_test.values[last_hours],
+    "x-",
+    alpha=0.2,
+    label="Actual demand",
+    color="black",
+)
+ax.plot(
+    median_predictions[last_hours],
+    "^-",
+    label="GBRT median",
+)
+ax.plot(
+    mean_predictions[last_hours],
+    "x-",
+    label="GBRT mean (Poisson)",
+)
+ax.fill_between(
+    np.arange(96),
+    percentile_5_predictions[last_hours],
+    percentile_95_predictions[last_hours],
+    alpha=0.3,
+    label="GBRT 90% interval",
+)
+_ = ax.legend()
+
+# %%
+# Looking at the performance of non-linear regression models vs
+# perfect models:
+fig, axes = plt.subplots(ncols=3, figsize=(15, 6), sharey=True)
+fig.suptitle("Non-linear regression models")
+predictions = [
+    median_predictions,
+    percentile_5_predictions,
+    percentile_95_predictions,
+]
+labels = [
+    "Median",
+    "5th percentile",
+    "95th percentile",
+]
+for ax, pred, label in zip(axes, predictions, labels):
+    ax.scatter(pred, y_test.values, alpha=0.3, label=label)
+    ax.plot([0, y.max()], [0, y.max()], "--", label="Perfect model")
+    ax.set(
+        xlim=(0, y.max()),
+        ylim=(0, y.max()),
+        xlabel="Predicted demand",
+        ylabel="True demand",
+    )
+    ax.legend()
+
+plt.show()
+
+# %%
+# Concluding Remarks
+# ------------------
+# Through this example we explored time series forecasting using lagged
+# features. We compared a naive regression (using the standardized
+# `train_test_split`) with a proper time series evaluation strategy using
+# `TimeSeriesSplit`. We observed that the model trained using `TimeSerieSplit`
+# gives a better Mean Average Percentage Error (MAPE) rate than the naive
+# method. We also analyzed the predictive uncertainty of our model via

```

Comment:
Yes and in particular, it's not "better". The two MAPE numbers cannot be compared because they are not evaluated on the sample test sets.

It's the shuffle-split procedure that is not reflecting the true setting and as a result produces an overly optimistic MAPE number.